src/gymnasium_interface/task_executor.py

```diff
@@ -0,0 +1,111 @@
+from pycram.worlds.bullet_world import BulletWorld
+from pycram.world_concepts.world_object import Object
+from pycram.datastructures.enums import ObjectType, WorldMode, Grasp
+from pycram.datastructures.pose import Pose
+from pycram.designators.action_designator import NavigateAction, PickUpAction, PlaceAction, OpenAction, CloseAction
+from pycram.designators.object_designator import BelieveObject
+from pycram.process_module import simulated_robot
+
+class PyCRAMTaskExecutor:
+    def __init__(self):
+        """Initializes task executor for PyCRAM actions."""
+        self.world = BulletWorld(WorldMode.GUI)
+        self.robot = None
+        self.apartment = None
+
+    def clear_world(self):
+        """Removes all objects from the BulletWorld."""
+        print("Clearing all objects from BulletWorld...")
+        for obj in list(self.world.objects):
+            obj.remove()
+        print("All objects removed from BulletWorld.")
+
+    def reset_task(self, objects):
+        """Resets the simulation environment dynamically."""
+        self.clear_world()
+
+        # Reload the apartment URDF
+        self.apartment = Object("apartment", "environment", "apartment.urdf")
+
+        # Reinitialize the robot
+        self.robot = Object("pr2", ObjectType.ROBOT, "pr2.urdf", pose=Pose([1.2, 1, 0]))
```

Comment:
Would be cooler to make this dependend on the robot that was used before instead of always using the Pr2